### PR TITLE
MM-49746 - Calls: Show call thread title in root calls post

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Show call thread title in root calls post

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-49746

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Android: 13, Pixel 6
- iOS: 15.7, iPhone 7 plus

#### Screenshots
image

#### Release Note
```release-note
Calls: Show the call thread in root calls posts.
```

### CODE REVIEW

File: app/products/calls/components/calls_custom_message/calls_custom_message.tsx [hunk 1]
```diff
@@ -133,97 +138,108 @@ export const CallsCustomMessage = ({
...
-                    size={16}
-                    style={style.phoneHangupIcon}
-                />
-                <View style={style.messageText}>
-                    <FormattedText
-                        id={'mobile.calls_call_ended'}
-                        defaultMessage={'Call ended'}
-                        style={style.startedText}
+            <>
+                {title}
```
Comment (larkox) [inline app/products/calls/components/calls_custom_message/calls_custom_message.tsx:149]: Sometimes react native gets a bit picky on what you pass in these cases (passing stuff that is not a boolean for example). Have you tested this properly works when the title is empty? If it works well, no changes needed :smile:

File: app/products/calls/components/calls_custom_message/calls_custom_message.tsx [hunk 2]
```diff
@@ -133,97 +138,108 @@ export const CallsCustomMessage = ({
...
-                        }
-                        <Text style={style.separator}>{'•'}</Text>
-                        <FormattedText
-                            id={'mobile.calls_lasted'}
-                            defaultMessage={'Lasted {duration}'}
-                            values={{duration: moment.duration(post.props.end_at - post.props.start_at).humanize(false)}}
-                            style={style.timeText}
-                        />
+                            <Text>{' '}</Text>
+                            {
```
Comment (larkox) [inline app/products/calls/components/calls_custom_message/calls_custom_message.tsx:169]: Why the brackets?

**Review by larkox — APPROVED** [review]
LGTM, Just two minor things.

**Comment by streamer45** [comment]
@cpoile Would it make sense to sync with @tanmay-des as well on this? I think there's a chance to possibly redesign a bit how the call title is going to look like. I know that on the web side it's not really matching the [latest design](https://www.figma.com/file/9HL2HADCQ0lLmn1RBOBGGV/Calls%3A-Recording?node-id=51%3A124715&t=oCZUNYRVwCLSpIH9-0) plus it comes with some limitations (e.g. it doesn't currently support at mentions). That said, I am not opposed to get this in and do further work later on but since we waited this long we may as well do it all once.

File: app/products/calls/components/calls_custom_message/calls_custom_message.tsx (hunk 1 above)
Comment (cpoile) [reply app/products/calls/components/calls_custom_message/calls_custom_message.tsx:150]: It works, but I changed it to be safer. 👍

**Comment by cpoile** [comment]
@streamer45 I got Tanmay's sign off on it, and we'll do the root post design in another effort.

**Review by streamer45 — APPROVED** [review]

**Pull request merged**